### PR TITLE
feat: support aten.index_select converter

### PR DESCRIPTION
**Pull request by chohk88**
# Description

New feature to support aten.index_select converter. I also add test case for different dimensions.

Fixes # (https://github.com/pytorch/TensorRT/issues/2708)

## Type of change

- New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project (You can use the linters)
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests to verify my fix or my feature
- [x] New and existing unit tests pass locally with my changes
- [x] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: py/torch_tensorrt/dynamo/conversion/impl/index.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    ctx: ConversionContext,
+    target: Target,
+    source_ir: Optional[SourceIR],
+    name: str,
+    input: TRTTensor,
+    dim: int,
+    index: TRTTensor,
+) -> TRTTensor:
+    input_tensor = get_trt_tensor(ctx, input, f"{name}_input")
+    index_tensor = get_trt_tensor(ctx, index, f"{name}_index")
```
[the remark this answers is not in the document]
Comment (chohk88) [reply py/torch_tensorrt/dynamo/conversion/impl/index.py]: I removed ITensor convert functions.

File: py/torch_tensorrt/dynamo/conversion/impl/index.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    ctx: ConversionContext,
+    target: Target,
+    source_ir: Optional[SourceIR],
+    name: str,
+    input: TRTTensor,
+    dim: int,
+    index: TRTTensor,
+) -> TRTTensor:
+    # The axis parameter specifies the dimension along which to index.
+    gather_layer = ctx.net.add_gather(input, index, axis=dim)
```
Comment (gs-olive) [inline py/torch_tensorrt/dynamo/conversion/impl/index.py:20]: `dim` likely needs to be corrected using `get_positive_dim` to ensure the value is positive for `add_gather`

File: tests/py/dynamo/conversion/test_index_select_aten.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+
+class TestIndexSelectConverter(DispatchTestCase):
+    @parameterized.expand(
+        [
+            ("1d_input", (10,), 0, (1,)),
+            ("2d_input_dim_0", (10, 3), 0, (0, 2)),
+            ("2d_input_dim_1", (5, 10), 1, (1, 2, 3)),
+            ("3d_input_dim_0", (10, 5, 10), 0, (0, 5)),
+            ("3d_input_dim_2", (10, 5, 10), 2, (3, 3, 4)),
```
Comment (gs-olive) [inline tests/py/dynamo/conversion/test_index_select_aten.py:16]: Add a test case for a negative `dim` input

File: py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py [hunk 4]
```diff
@@ -2782,3 +2782,28 @@ def aten_ops_roll(
...
+    }
+)
+def aten_ops_index_select(
+    ctx: ConversionContext,
+    target: Target,
+    args: Tuple[Argument, ...],
+    kwargs: Dict[str, Argument],
+    name: str,
+) -> Union[TRTTensor, Sequence[TRTTensor]]:
+    return impl.index.index_select(
```
Comment (zewenli98) [inline py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py:2801]: It seems that the `index_select` function could be put into `select.py`

File: py/torch_tensorrt/dynamo/conversion/impl/index.py (hunk 2 above)
Comment (chohk88) [reply py/torch_tensorrt/dynamo/conversion/impl/index.py]: I have modified it. Thanks!

File: tests/py/dynamo/conversion/test_index_select_aten.py (hunk 3 above)
Comment (chohk88) [reply tests/py/dynamo/conversion/test_index_select_aten.py:17]: I have added a test case for a negative `dim` input and verified a test case. Thank you!

File: py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py (hunk 4 above)
Comment (chohk88) [reply py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py]: I moved `index_select` inside `select.py`. Thank you!

**Review by zewenli98 — APPROVED** [review]
LGTM!

File: py/torch_tensorrt/dynamo/conversion/impl/__init__.py [hunk 5]
```diff
@@ -12,6 +12,7 @@
     elementwise,
     embedding,
     grid,
+    index,
```
Comment (gs-olive) [inline py/torch_tensorrt/dynamo/conversion/impl/__init__.py:15]: This can likely be removed - it seems to be causing a circular import error in CI

Comment (chohk88) [reply py/torch_tensorrt/dynamo/conversion/impl/__init__.py]: Thanks! It seems I overlooked removing an unnecessary import.

**Review by gs-olive — APPROVED** [review]
Looks good to me!
